specs/searchable-asn-group/searchable-asn-group.md

```diff
@@ -0,0 +1,112 @@
+# Searchable ASN Filters (ff-asn-group)
+
+Some data sets have huge amounts of filters per filter group (150+). Too many to display all at once and too many to browse by eye.
+
+To improve usability the `ff-asn-group` gets an _optional_ search field. It is scoped to its associated filter group.
+
+
+## General
+
+This feature is an addition to the existing `ff-asn-group`. HTML examples are limited to new parts and minimal context for focus.
+
+The new search field is always the first element inside the collapsible `.ffw-wrapper` container. This means the filter group must be expanded in order to use it.
+
+Users may specify a custom search field template. It is defined via the `slot="filterSearch"` attribute. As it is a slot, its location inside the `ff-asn-group` template is irrelevant.
+
+## Features
+
+#### search as you type
+
+Filter items are searched for while user is typing in search field. While search term is present, filter group is kind of in _"search mode"_ compared to _"regular mode"_.
+
+- matching with `new Regexp(searchTerm, "gi")` or better algorithm
+- while search term is present, _"show more"_ disappears, matches of detailed and hidden links are merged into one list
+- not matching items are hidden
+- order of matched results remains unchanged compared to order in original FF response
+- no special treatment for already selected filters
+  - selected filters are filtered just like all others
+
+If a filter item (selected or unselected) is clicked while in _"search mode"_, the search field is cleared and the component returns to _"regular mode"_.
+
+
+## Default HTML
+
+The search field is only rendered if the attribute `searchable-from` is set with a valid number on `ff-asn-group`.
+
+If `searchable-from` is not set, the search field will not be rendered.
+
+#### Setup
+```html
+<ff-asn-group searchable-from="31">
+</ff-asn-group>
+```
+
+#### Rendered
+
+Assuming there are `31` or more links (_detailed_ and _hidden_ combined) in this group.
+
+```html
+<ff-asn-group searchable-from="31">
+    <div class="ffw-asn-group-container">
+        <div class="ffw-wrapper">
+
+            <div slot="filterSearch"><input></div>  <!-- this is the new part -->
+
+            <div data-container="detailedLinks">
+            ...
+```
+
+_Note that the search field is always the first item inside `.ffw-wrapper`._
+
+
+## User specified templates
+
+User templates are specified through the `slot="filterSearch"` attribute.
+
+The `input` element is mandatory. An exception is thrown if it is not provided.
+
+#### Setup
+
+```html
+<ff-asn-group for-group="Manufacturer" searchable-from="31">
```

Comment:
Would this configuration by attribute also be possible via FACT-Finder UI and the configuration of filters? Tying this setting within the code to specific groups doesn't seem to be flexible enough.

Reply:
To make it configurable in the FACT-Finder UI we can issue a feature request to FF-Dev.
But I can't tell how soon (or if at all) they will be able to implement it.

Would you prefer to not configure this value in code at all?
How much flexibility would be reasonable?

If maintenance of the threshold value across multiple groups is a concern, we could move the attribute to `ff-asn` and have it propagate to all groups. That is, if individual threshold settings aren't needed. Otherwise we could make the attribute available on both `ff-asn` and `ff-asn-group`.
This could look something like this:
```html
<ff-asn searchable-from="31">
  <ff-asn-group for-group="Manufacturer"></ff-asn-group>
  <ff-asn-group for-group="Recommended Use"></ff-asn-group>
</ff-asn>
```

Reply:
Yes, from my point of view this belongs to the filter configuration in FF UI. But I understand that implementing new stuff there might take long time.

The point is that you are free to reconfigure what filters are used and shown via the FF UI. Such changes then might need a deployment to update the code base accordingly. This feels not right.

Your suggestion looks like a reasonable compromise. Setting the threshold at `<ff-asn>` is a good idea. Then I think it is good to activate the search field for individual groups, e.g. brand.

```html
<ff-asn searchable-from="31">
  <ff-asn-group for-group="Brand" searchable></ff-asn-group>
  ...
<//ff-asn>
```

Reply:
I'm currently talking to FACT-Finder Dev about a possible feature request. I feel we should explore possibilities of a UI implementation before settling on a compromise to avoid incompatibilities with a yet to be specified FF feature.

Reply:
When setting `searchable-from`, I would assume that all groups with the required amount of elements shall be searchable by default. Thus favouring an opt-out approach for individual groups' searchability to reduce the number of attributes used.
```html
<ff-asn searchable-from="31">
  <ff-asn-group for-group="Brand"></ff-asn-group>
  <ff-asn-group for-group="Colour" not-searchable></ff-asn-group>
  <ff-asn-group for-group="Usage"></ff-asn-group>
```
Or would you say it is more likely that a group exceeding the threshold shall **not** be searchable?

Reply:
I like your approach more!